### PR TITLE
Add syntax highlighting for Roc docs

### PR DESCRIPTION
**Pull request by lukewilliamboswell**
Adds syntax highlighting for the documentation `roc docs`. Removes some unused code. Refactors the html `highlight` into a crate that is common to `static-site` example platform and the `roc_docs` crate.

Note I haven't completed a `cargo test` run yet, but have manually tested this doesn't break website docs, or tutorial.

`.css` styles for code samples copied from tutorial.

Screen Shot 2023-03-06 at 22 44 45

### CODE REVIEW

**Comment by lukewilliamboswell** [comment]
@Anton-4 updated this PR to support fenced code as discussed on [zulip](https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/code.20in.20docs/near/340052655). I left the actual checking and REPL with TODO's for now. I also updated all of the Tutorial markdown to the new style and manually checked each example. There are a lot of examples that need to be touched up to make more semantic sense, but that is out of scope for this PR.

**Review by Anton-4 — CHANGES_REQUESTED** [review]

File: crates/compiler/builtins/roc/Num.roc [hunk 1]
```diff
@@ -1390,7 +1392,7 @@ toF64Checked : Num * -> Result F64 [OutOfBounds]
 # ## If `max` is less than `min`, then first the number will be truncated to `max`
 # ## digits, and then zeroes will be added afterwards until it reaches `min` digits.
 # ##
-# ## >>> Num.format 1.23 { decPlaces: 0, decPointVis: AlwaysShow }
+# ##     Num.format 1.23 { decPlaces: 0, decPointVis: AlwaysShow }
```
[the remark this answers is not in the document]
Comment (lukewilliamboswell) [reply crates/compiler/builtins/roc/Num.roc:1395]: This while section was weird. It was like this feom a previous PR. I'm not sure but it looks like we've commented all this section out for some reason? Should we just remove the code, or do we have plans to add it back in?

[the remark this answers is not in the document]
Comment (Anton-4) [reply crates/compiler/builtins/roc/Num.roc:1395]: Oh right it's just commented out, let's remove it, there's no TODO and we'll know where to find it if we want it back.

**Comment by lukewilliamboswell** [comment]
Updated to improve token groups as discussed on [zulip](https://roc.zulipchat.com/#narrow/stream/304641-ideas/topic/code.20in.20docs/near/340394575). I experimented with different variations, and this is what I like the most.

I also added a colour scheme which looks nice for both light and dark modes. It uses five colours;

- Cyan for Number, String, Tag, Type literals
- Magenta for Keywords and punctuation
- Orange for Operators
- Violet for Variables modules and field names
- Green for Types, Tags, and Modules

## Light Example

Screen Shot 2023-03-10 at 11 39 49

## Dark Example

Screen Shot 2023-03-10 at 11 39 59

**Comment by rtfeldman** [comment]
Looks cool!

One thing I should have mentioned earlier: since these are docs for everyone (as opposed to an individual's personal editor preference), it's important that we take accessibility into account. I'm specifically thinking contrast ratios, which at a glance look probably too low to pass WCAG AA accessibility conformance at the moment.

When I picked the theme colors for the tutorial and docs I used [this tool](https://accessible-colors.com/)) to get the colors conformant, and it was very helpful!

(There's [some debate](https://uxmovement.com/buttons/the-myths-of-color-contrast-accessibility/) as to whether WCAG AA is optimal for darker text on colored backgrounds in particular, but that probably won't come up here.)

**Comment by lukewilliamboswell** [comment]
@rtfeldman I tweaked the colours they should all now pass WCAG AA compliant for both Light and Dark modes. Note I only checked the code snippets and have left the other colours as they were. I tried to clean up some of the unused css along the way, and unify more of the styles between docs and tutorial. There is still more work there to do, but that it out of scope for this PR.

File: crates/compiler/builtins/roc/Num.roc (hunk 1 above)
[the remark this answers is not in the document]
Comment (lukewilliamboswell) [reply crates/compiler/builtins/roc/Num.roc]: I removed all of the commented code.

**Review by Anton-4 — APPROVED** [review]

File: www/generate_tutorial/src/input/tutorial.md [hunk 2]
```diff
@@ -161,7 +161,7 @@ main =
     Stdout.line "There are \(total) animals."
 ```
 
-Now run `roc dev` again. This time the "Downloading …" message won't appear; the file has been cached from last time, and won't need to be downloaded again.
+Now run `roc dev` again. This time the "Downloading ..." message won't appear; the file has been cached from last time, and won't need to be downloaded again.
```
Comment (Anton-4) [inline www/generate_tutorial/src/input/tutorial.md:164]: Did the ellipsis symbol cause any issues?

**Review by Anton-4 — CHANGES_REQUESTED** [review]

File: www/generate_tutorial/src/input/tutorial.md (hunk 2 above)
Comment (lukewilliamboswell) [reply www/generate_tutorial/src/input/tutorial.md:164]: Yes, it crashes the parser from memory. Was going to put an issue in to track it.

**Review by Anton-4 — APPROVED** [review]

**Comment by Anton-4** [comment]
Thank you very much @lukewilliamboswell! This was a big effort :muscle:

**Comment by rtfeldman** [comment]
@lukewilliamboswell hm, it looks like the colors on the code snippets don't pass WCAG AA 😅 

e.g. here's `getRatings`

Screen Shot 2023-0311 at 5 00 04 PM

Screen Shot 2023-03-11 at 4 59 12 PM

Would you mind taking a second pass at the various foreground/background combos in light and dark mode with [accessible-colors.com](accessible-colors.com)? It has some quick fix suggestions that I've found helpful in the past, e.g. changing the background color to #111423 might address more than one contrast issue here. 😄

**Comment by rtfeldman** [comment]
Actually, darkening the background that much makes it kinda hard to see that code snippets have a different background at all...

Screen Shot 2023-03-11 at 5 08 43 PM

...so maybe a combination of lightening the text a bit and darkening the background a bit? I often try these tweaks by changing from RGB to HSL representation and then changing just L to make it lighter or darker. (Can always change it back to a hex representation afterwards if desired.)

**Pull request merged**